src/dataflow/components/dataflow-program.tsx

```diff
@@ -532,4 +539,21 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
     }
   }
 
+  private zoomIn = () => {
+    const { k } = this.programEditor.view.area.transform;
+    this.setZoom(Math.min(MAX_ZOOM, k + .05));
+  }
+
+  private zoomOut = () => {
+    const { k } = this.programEditor.view.area.transform;
+    this.setZoom(Math.max(MIN_ZOOM, k - .05));
+  }
+
+  private setZoom = (zoom: number) => {
+    this.programEditor.view.area.transform = {k: zoom, x: 0, y: 0};
```

Comment:
This snaps the diagram to (0, 0), which is a little unexpected for zooming.

Playing with it, it was tricky to get it to zoom into the center, but this change at least keeps the top left corner where it was originally, which feels a lot smoother:

```
const currentTransform = this.programEditor.view.area.transform;
this.programEditor.view.area.transform = {k: zoom, x: currentTransform.x, y: currentTransform.y};
```

Reply:
Agreed.  That does feel much better.  Fixed.